### PR TITLE
Directly store needed attributes in `TreeHeadNode`

### PR DESCRIPTION
**Pull request by vepadulano**
This PR addresses the TODO comments to remove the superfluous public getters / properties logic in `TreeHeadNode`. Currently, just parses the user arguments in the `__init__` method and stores the needed attributes. My idea is that in the end we'll only need to store `npartitions` and `tree` (which are always needed) plus `defaultbranches` and `friendinfo` as "optional" class attributes (meaning they are initialized to `None` and changed if some conditions apply). 

For simplicity, in this PR I still leave `treename` and `inputfiles` attributes which are needed in the `Ranges.get_clusters` function.

The next PRs will address:
* The assumption we only have one unique treename in the RDataFrame
* Change in `Ranges.get_clusters` to accept a `TTree` instance as argument, pairing it with new C++ functions in `ROOT::Internal::TreeUtils` to retrieve a vector of clusters (plus some other metadata) depending if it's a TTree or TChain

If the logic for this PR is approved I'll add more docs and commit messages

### CODE REVIEW

File: bindings/experimental/distrdf/python/DistRDF/HeadNode.py [hunk 1]
```diff
@@ -131,6 +131,8 @@ def __init__(self, npartitions, *args):
         if isinstance(args[0], ROOT.TTree):
             # RDataFrame(tree, defaultBranches = {})
             self.tree = args[0]
+            # Retrieve information about friend trees when user passes a TTree
+            # or TChain object.
```
Comment (eguiraud) [inline bindings/experimental/distrdf/python/DistRDF/HeadNode.py:135]: this comment belongs to the previous commit

File: bindings/experimental/distrdf/test/test_headnode.py [hunk 2]
```diff
@@ -209,6 +225,11 @@ def test_num_entries_with_ttree_arg(self):
             v[i] = 1  # Change the vector element to 1
             tree.Fill()  # Fill the tree with that element
 
+        f.Write()
+
         hn = create_dummy_headnode(tree)
 
-        self.assertEqual(hn.get_num_entries(), 4)
+        self.assertEqual(hn.tree.GetEntries(), 4)
```
Comment (eguiraud) [inline bindings/experimental/distrdf/test/test_headnode.py:232]: what are you testing here?

**Review by eguiraud — APPROVED** [review]
LGTM

File: bindings/experimental/distrdf/test/test_headnode.py (hunk 2 above)
Comment (vepadulano) [reply bindings/experimental/distrdf/test/test_headnode.py:232]: This and the above tests in class `NumEntriesTest` were previously testing that the method `get_num_entries` of `TreeHeadNode` was returning the correct amount. Now, they test that the attribute `tree` of the class is the correct TTree/TChain that we expect by retrieving the number of entries.

Comment (vepadulano) [reply bindings/experimental/distrdf/test/test_headnode.py:232]: I may agree that it's not a super useful test, but I didn't want to remove it as a first step

**Comment by vepadulano** [comment]
* Removed unused import
* Updated class docstrings
* Added commit message
* Rebased

File: bindings/experimental/distrdf/python/DistRDF/HeadNode.py [hunk 3]
```diff
@@ -95,19 +94,35 @@ def build_ranges(self):
 class TreeHeadNode(Node.Node):
     """
     The head node of a computation graph where the RDataFrame data source is
-    a TTree. This head node is responsible for the following RDataFrame constructor::
+    a TTree or a TChain. This head node is responsible for the following
+    RDataFrame constructors::
```
Comment (etejedor) [inline bindings/experimental/distrdf/python/DistRDF/HeadNode.py:98]: Double colon

**Review by etejedor — APPROVED** [review]

File: bindings/experimental/distrdf/python/DistRDF/HeadNode.py (hunk 3 above)
Comment (vepadulano) [reply bindings/experimental/distrdf/python/DistRDF/HeadNode.py:98]: Double colon is how Sphinx parses a code block (doxygen would expect a ~~~{.py}) . It's still here and there in DistRDF, when we'll move to fully utilise doxygen will be changed

**Pull request merged**